webrtc/dtls/dtls_connection.nim

```diff
@@ -0,0 +1,222 @@
+# Nim-WebRTC
+# Copyright (c) 2024 Status Research & Development GmbH
+# Licensed under either of
+#  * Apache License, version 2.0, ([LICENSE-APACHE](LICENSE-APACHE))
+#  * MIT license ([LICENSE-MIT](LICENSE-MIT))
+# at your option.
+# This file may not be copied, modified, or distributed except according to
+# those terms.
+
+import chronos, chronicles
+import ../errors, ../stun/[stun_connection]
+
+import mbedtls/ssl
+import mbedtls/ssl_cookie
+import mbedtls/ssl_cache
+import mbedtls/pk
+import mbedtls/md
+import mbedtls/entropy
+import mbedtls/ctr_drbg
+import mbedtls/rsa
+import mbedtls/x509
+import mbedtls/x509_crt
+import mbedtls/bignum
+import mbedtls/error
+import mbedtls/net_sockets
+import mbedtls/timing
+
+logScope:
+  topics = "webrtc dtls_conn"
+
+# -- DtlsConn --
+# A Dtls connection to a specific IP address recovered by the receiving part of
+# the Udp "connection"
+
+type
+  MbedTLSCtx* = object
+    ssl*: mbedtls_ssl_context
+    config*: mbedtls_ssl_config
+    cookie*: mbedtls_ssl_cookie_ctx
+    cache*: mbedtls_ssl_cache_context
+    timer*: mbedtls_timing_delay_context
+    pkey*: mbedtls_pk_context
+    srvcert*: mbedtls_x509_crt
+
+    ctr_drbg*: mbedtls_ctr_drbg_context
+    entropy*: mbedtls_entropy_context
+
+  DtlsConn* = ref object
+    conn*: StunConn # The wrapper protocol Stun Connection
+    laddr: TransportAddress # Local address
+    raddr*: TransportAddress # Remote address
+    dataRecv: seq[byte] # data received which will be read by SCTP
+    sendFuture: Future[void].Raising([CancelledError, WebRtcError])
+    # This future is set by synchronous Mbed-TLS callbacks and waited, if set, once
+    # the synchronous functions ends
+
+    # Close connection management
+    closed: bool
+    closeEvent: AsyncEvent
+
+    # Local and Remote certificate, needed by wrapped protocol DataChannel
+    # and by libp2p
+    localCert*: seq[byte]
+    remoteCert*: seq[byte]
+
+    # Mbed-TLS contexts
+    ctx*: MbedTLSCtx
```

Comment:
I don't think this should be public, it is an implementation detail.

Reply:
I still see it as public.

Reply:
That's my bad. I privatized all MbedTLSCtx fields and forgot to set this to private, it's really done now :sweat_smile: